plugins/amazonq/codetransform/jetbrains-community/tst/software/aws/toolkits/jetbrains/services/codemodernizer/CodeWhispererCodeModernizerUtilsTest.kt

```diff
@@ -87,4 +90,23 @@ class CodeWhispererCodeModernizerUtilsTest : CodeWhispererCodeModernizerTestBase
         assertThat(expected).isEqualTo(mutableList)
         verify(clientAdaptorSpy, times(2)).getCodeModernizationJob(any())
     }
+
+    @Test
+    fun `getTableMapping on complete step 0 progressUpdates creates map correctly`() = runBlocking {
+        val jobStats = "{\"name\":\"Job statistics\", \"columnNames\":[\"name\",\"value\"],\"rows\":" +
+            "[{\"name\":\"Dependencies to be replaced\",\"value\":\"5\"},{\"name\":\"Deprecated code instances to be replaced\",\"value\":\"10\"}]}"
+        val depChanges = "{\"name\":\"Dependency changes\", \"columnNames\":[\"dependencyName\",\"action\",\"currentVersion\",\"targetVersion\"],\"rows\":" +
+            "[{\"dependencyName\":\"org.springboot.com\",\"action\":\"Update\",\"currentVersion\":\"2.1\",\"targetVersion\":\"2.4\"}]}"
+        val apiChanges = "{\"name\":\"Deprecated API changes\", \"columnNames\":[\"apiFullyQualifiedName\",\"numChangedFiles\"],\"rows\":" +
+            "[{\"apiFullyQualifiedName\": \"java.lang.bad()\", \"numChangedFiles\": \"3\"}]}"
+        val fileChanges = "{\"name\":\"File changes\", \"columnNames\":[\"relativePath\",\"action\"],\"rows\":" +
+            "[{\"relativePath\":\"pom.xml\",\"action\":\"Update\"}, {\"relativePath\":\"src/main/java/BloodbankApplication.java\",\"action\":\"Update\"}]}"
```

Comment:
ideally you would use a multiline string so that these would actually be readable while skimming through test cases